addons/website_membership/controllers/main.py

```diff
@@ -100,12 +117,21 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=0, *
         if free_partner_ids:
             memberships_data.append({'id': 'free', 'name': _('Free Members')})
             if not membership_id or membership_id == 'free':
-                memberships_partner_ids['free'] = free_partner_ids
+                if count_members < offset + limit:
+                    memberships_partner_ids['free'] = free_partner_ids[offset-count_members:offset+limit-count_members]
+                showed_free_partner_ids = free_partner_ids[:2000-len(all_partner_ids)]
+                all_partner_ids |= set(showed_free_partner_ids)
```

Comment:
`set` doesn't respect the orders
Therefore, all_partner_ids is not ordered anymore. That is not such an issue, but the partners displayed on the Google Maps will be displayed randomly, and possibly only the free members will be displayed, while the gold/silver/bronze members should have the priority :s.